### DISCUSSION TITLE
Socialite OAuth 2.0 for Twitter

### DISCUSSION THREAD

**Discussion by MrMooky** [laravel/framework#40777] [Ideas]
Since Twitter now uses OAuth 2.0, the Laravel Socialite Facade should be updated to make use of that and also allow the `stateless()` method.

From the [docs](https://laravel.com/docs/8.x/socialite#stateless-authentication):

> Stateless authentication is not available for the Twitter driver, which uses OAuth 1.0 for authentication.

**Comment by danielpetrica** [comment]
I may be wrong, but it seems to me that the OAuth 2.0 flow is already there, it's just the stateless method that is missing as I've tried adding the `'oauth'=>2` config in services.php, and it uses the new OAuth flow and the token returned can be used to access the v2 endpoints

**Reply by danielpetrica** [reply]
I'm sorry, it seems I'm wrong. I couldn't set the scopes correctly in the permission request.

**Reply by danielpetrica** [reply]
Ok, I was right the first time scopes and stateless can be configured using the methods from the socialite abstract class.
In the redirect : ` return Socialite::driver('twitter')->stateless()->setScopes(config('services.twitter.scopes'))->redirect();`

and in the callback `$user = Socialite::driver('twitter')->stateless()->user();`
